### DISCUSSION TITLE
Safewatch Pro 2000 with level shifters

### DISCUSSION THREAD

**Discussion by crstrand** [Dilbert66/esphome-vistaECP#75] [Q&A]
My alarm has VISTA-20 on an label on the PCB.  It is using a wireless extender that is / should be configured for two fobs and one wireless motion sensor.
I modified the YAML to reflect my system setup (I think) and uploaded the code to my nodemcu-32s.
The status window shows me a lot of
`[20:06:47][I][CMD:445]: F8 78 06 7E 00 00 00 00 00 00 00 00 77 `
`[20:06:47][I][EXT:445]: F6 78 00 00 FF BF FF FF FF FF FF FF FF `
type of messages, but the text_sensor messages don't reflect changes in the system (open the door, and its status does not change).
I'm using a level shifter board that I bought from AliExpress.  For now I'm just using red, black, and yellow wires and feeding green into the green monitor pin.  I had green connected to the output pin on the ESP32 (GPIO21), but I got zone 9 faults (I don't have 9 zones).
How do I start debugging this thing?
![ECPInterface_level_shifter](https://user-images.githubusercontent.com/2912270/168203362-a5ea8a22-d91d-4eb9-b5cf-660833fb5b82.png)

**Comment by Dilbert66** [comment]
You should be seeing f7 messages.  That's the cmd that affects zone statuses.   I believe the safewatch pro 2000 is actually a vista20se and is not the same as a vista-20p so won't work.  You need to use the vista20se code in the src directory for this system.

**Reply by crstrand** [reply]
I have attached the log I am getting with the Vista20SE code.  No F7 to be had.
I did have one minor mod to vistaAlarm.h to get it to compile. alarmStatus should be alarmStatusType (line 255).
I see the 20SE baud rate is 2400 and the 20 is 4800.  Is that correct? Update: looking at the code it seems there is some automatic baud detection going on.  It starts out at 2400, but changes to 4800 under certain conditions.
I'm willing to be a 20SE tester, I just need some more insight on how this whole thing works. :)
[logs_esp-alarm-panel_logs.txt](https://github.com/Dilbert66/esphome-vistaECP/files/8686504/logs_esp-alarm-panel_logs.txt)

**Comment by Dilbert66** [comment]
Just use the files in the directory vista20se instead of the ones in vistaecpinterface.  The yaml remains the same.
The baud rate differs on the type of cmd that is sent as well as for key events. It's a much older system using an older protocol.

**Reply by crstrand** [reply]
That is what I did.  I copied the all the SE files to \\TOWER\appdata\ESPHome\vistaEcpInterface then did an install so it would compile and upload the new code.  I will try a clean and another install.
Same result after the clean and re-install.

**Comment by Dilbert66** [comment]
You're getting some odd traffic on the monitor line. I suspect there is something odd there with the wiring.  The yellow line (cmd:xxx) line is fine as I can see the panel traffic.  It's the green line monitor that is generate garbage traffic. (EXT:xxxx lines).  FYI, there are no F7 lines for the SE.  The panel messages com on the FE and FF lines.  The EXT logs should never have FF traffic so for sure, you need to look at that wiring.  I'm not familiar with those level shifters but looking at how you have it wired I can see the problem.  You are tying the green line high to red via a pullup resistor.  That's a no no.  The green line should be idle low. so you are forcing invalid data on it.

**Reply by crstrand** [reply]
I hooked up the circuit using some PC817 optocouplers I had laying around.  They claim to have a ratio of 50.
The garbage is gone and I seem to be seeing come "normal" status messages, but when I open a window to trip a zone, it doesn't detect anything
[logs_esp-alarm-panel_upload.txt](https://github.com/Dilbert66/esphome-vistaECP/files/8750797/logs_esp-alarm-panel_upload.txt)
.

**Reply by crstrand** [reply]
I need to go through my optocoupler circuit.  It must not be correct.
I went super simple and used the 33/10K resistor divider on the yellow line only.
I'm now getting the following log.  During this log I should have tripped the Basement window (south) zone/sensor
[logs_esp-alarm-panel_logs_yellow_only_with_resistors.txt](https://github.com/Dilbert66/esphome-vistaECP/files/8751089/logs_esp-alarm-panel_logs_yellow_only_with_resistors.txt)
.

**Comment by Dilbert66** [comment]
Yes, you have an issue with the optocoupler version as it has no data traffic.  The second version looks ok. I don't see any fault messages but since the FE's are correctly decoded, then any faults would show in those messages.

**Reply by crstrand** [reply]
I think the window was open around the time of the "DISARMED-Press * to show faults" message.
What are the 00 00  and 00 10 messages for?
Similarly for:
FB 02 20 81 62
FB 02 25 60 7E
FB 02 25 82 5C

If there's a link where I can glean this information for myself, I'm more than happy to read. :)

**Comment by Dilbert66** [comment]
FB are the RF module messages. There are no links to information. I had to figure it out by analyzing the packets with a logic analyzer.  You can look at the vista.cpp file for more details on the decoding.

As to the 00 10 messages, those are zone/system status flags.
